src/index.ts

```diff
@@ -33,28 +34,36 @@ namespace CommandIDs {
  * A service providing variable introspection.
  */
 const variableinspector: JupyterFrontEndPlugin<IVariableInspectorManager> = {
-  id: '@lckr/jupyterlab_variableinspector',
-  requires: [ICommandPalette, ILayoutRestorer, ILabShell],
+  id: '@lckr/jupyterlab_variableinspector:plugin',
+  requires: [ICommandPalette, ILayoutRestorer, ILabShell, ISettingRegistry],
   provides: IVariableInspectorManager,
   autoStart: true,
-  activate: (
+  activate: async (
     app: JupyterFrontEnd,
     palette: ICommandPalette,
     restorer: ILayoutRestorer,
-    labShell: ILabShell
-  ): IVariableInspectorManager => {
+    labShell: ILabShell,
+    settingRegistry: ISettingRegistry
+  ): Promise<IVariableInspectorManager> => {
     const manager = new VariableInspectorManager();
     const category = 'Variable Inspector';
     const command = CommandIDs.open;
     const label = 'Open Variable Inspector';
     const namespace = 'variableinspector';
     const tracker = new WidgetTracker<VariableInspectorPanel>({ namespace });
+    let settings: ISettingRegistry.ISettings;
+
+    try {
+      settings = await settingRegistry.load(variableinspector.id);
+    } catch (error) {
+      console.error('Failed to load settings for the Git Extension', error);
+    }
```

Comment:
Good practice wants that asynchronous code should not block the activation of the plugin as it may delay the full application start up.

A better pattern is therefore to use a Promise approach:

```suggestion
    Promise.all([settingRegistry.load(variableinspector.id), app.restored])
    .then(settings => {
      // ... here use the settings
    })
    .catch(error => {
      console.error('Failed to load settings for the Git Extension', error);
    });
```